include/jemalloc/internal/edata.h

```diff
@@ -13,6 +13,12 @@
 #include "jemalloc/internal/sz.h"
 #include "jemalloc/internal/typed_list.h"
 
```

Comment:
> one of which is disallowed by anyone but the emap

That is true. I'm with you that we should have one copy of metadata instead of cached / duplicated. The only / main advantage of that is probably convenience.

How about we remove the duplicated states in the next PR? There will be quite some code motion for that change. This one is getting pretty large, and I already have a few things want to do but not included (e.g. the metadata_brief we also discussed).

Reply:
Yep, totally fine.